source/rust_verify/src/verifier.rs

```diff
@@ -512,8 +513,15 @@ impl Verifier {
         let report_long_running = || {
             let mut counter = 0;
             let report_fn: Box<dyn FnMut(std::time::Duration) -> ()> = Box::new(move |elapsed| {
-                let msg =
+                let mut msg =
                     format!("{} has been running for {} seconds", context.1, elapsed.as_secs());
+                if let Some(in_line_order) = reports_in_line_order.take() {
```

Comment:
Should we also sort the existing reports here (i.e. sort the current contents of `in_line_order`)?  I realize that there may still be reports printed later out of order, but it seems like the more often we sort, the closer we'll be to source order.